crates/task-impls/src/quorum_proposal/handlers.rs

```diff
@@ -120,31 +122,38 @@ async fn visit_leaf_chain<TYPES: NodeType, I: NodeImplementation<TYPES>>(
 
     // The next view is the next view we're going to traverse, it is the validated state of the
     // parent of the proposal.
-    let next_view = validated_state_map
+    let parent_view = validated_state_map
```

Comment:
To avoid the check on line 144, would it be better to make this a while loop and move the while loop on line 140 inside here? Something like:
```rust
let mut view_number = proposal_parent_view_number;
while let Some(view) = validated_state_map.get(&view_number) {
  let leaf_commitment = view.leaf_commitment().context(...)?;
  let leaf = saved_leaves.get(&leaf_commitment).context(...)?;
  ...
  // Update view_number
}
```

I think this is clearer but I don't have a strong opinion, so it's up to you!

Reply:
I wrote a test to make sure I didn't break anything, added this in! Great suggestion.